Explorer/Assets/Scripts/ECS/Unity/AvatarShape/Systems/AvatarShapeHandlerSystem.cs

```diff
@@ -0,0 +1,96 @@
+using Arch.Core;
+using Arch.System;
+using Arch.SystemGroups;
+using Arch.SystemGroups.Throttling;
+using DCL.Diagnostics;
+using DCL.ECSComponents;
+using ECS.Abstract;
+using DCL.Utilities;
+using ECS.LifeCycle;
+using ECS.LifeCycle.Components;
+using ECS.Prioritization.Components;
+using ECS.Unity.AvatarShape.Components;
+using ECS.Unity.Groups;
+using ECS.Unity.Transforms.Components;
+
+namespace ECS.Unity.AvatarShape.Systems
+{
+    [UpdateInGroup(typeof(ComponentInstantiationGroup))]
+    [LogCategory(ReportCategory.AVATAR)]
+    [ThrottlingEnabled]
+    public partial class AvatarShapeHandlerSystem : BaseUnityLoopSystem, IFinalizeWorldSystem
+    {
+        private WorldProxy globalWorld;
+        private ReleaseOnEntityDestroy releaseOnEntityDestroy;
+
+        public AvatarShapeHandlerSystem(World world, WorldProxy globalWorld) : base(world)
+        {
+            this.globalWorld = globalWorld;
+            releaseOnEntityDestroy = new ReleaseOnEntityDestroy(globalWorld);
+        }
+
+        protected override void Update(float t)
+        {
+            LoadAvatarShapeQuery(World);
+            UpdateAvatarShapeQuery(World);
+
+            HandleComponentRemovalQuery(World);
+            HandleEntityDestructionQuery(World);
+        }
+
+        [Query]
+        [None(typeof(SDKAvatarShapeComponent))]
+        private void LoadAvatarShape(in Entity entity, ref PBAvatarShape pbAvatarShape, ref PartitionComponent partitionComponent, ref TransformComponent transformComponent)
+        {
+            Entity globalWorldEntity = globalWorld.Create(pbAvatarShape, partitionComponent, transformComponent);
+            World.Add(entity, new SDKAvatarShapeComponent(globalWorldEntity));
+        }
+
+        [Query]
+        private void UpdateAvatarShape(ref PBAvatarShape pbAvatarShape, ref SDKAvatarShapeComponent sdkAvatarShapeComponent)
+        {
+            if (!pbAvatarShape.IsDirty)
+                return;
+
+            globalWorld.Add(sdkAvatarShapeComponent.globalWorldEntity, pbAvatarShape);
```

Comment:
If `PbAvatarShape` was changed `World.Add` will throw an exception as the component is already added. You should differentiate between adding and setting

Reply:
Done!